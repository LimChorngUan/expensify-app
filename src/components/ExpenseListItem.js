import React from 'react'
import { Link } from 'react-router-dom'

const ExpenseListItem = ({
  id,
  description,
  amount,
  createdAt,
}) => (
  <div>
    <Link to={`/edit/${id}`}>{description}</Link>
    <p>Amount: {amount}</p>
    <p>Create at: {createdAt}</p>
  </div>
)

export default ExpenseListItem
