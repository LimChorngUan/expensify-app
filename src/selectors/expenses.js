import _ from 'lodash'

const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate}) => {
  const filteredExpenses = expenses.filter(expense => {
    const textMatch = expense.description.toLowerCase().includes(text.toLowerCase())
    const startDateMatch = typeof startDate !== `number` || expense.createdAt >= startDate
    const endDateMatch = typeof endDate !== `number` || expense.createdAt <= endDate

    return textMatch && startDateMatch && endDateMatch
  })

  if (sortBy === `amount`) {
    return _.sortBy(filteredExpenses, [`amount`])
  } else if (sortBy === `date`) {
    return _.sortBy(filteredExpenses, [`createdAt`])
  }
}

export default getVisibleExpenses