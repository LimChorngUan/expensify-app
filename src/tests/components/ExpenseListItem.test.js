import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ExpenseListItem from '../../components/ExpenseListItem'
import { expenses } from '../fixture/expenses'

describe(`Component: ExpenseListItem`, () => {
  const item = expenses[0]
  
  test(`Should render ExpenseListItem correctly`, () => {
    const wrapper = shallow(<ExpenseListItem {...item} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})