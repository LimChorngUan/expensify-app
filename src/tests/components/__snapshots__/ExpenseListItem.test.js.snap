// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ExpenseListItem Should render ExpenseListItem correctly 1`] = `
<div>
  <Link
    to="/edit/A"
  >
    A
  </Link>
  <p>
    Amount: 
    1
  </p>
  <p>
    Create at: 
    1
  </p>
</div>
`;
